Pilot/PlatformService.swift

```diff
@@ -17,50 +18,57 @@ class PlatformService: NSObject {
-          completion: { returnVideos in
-            let cloudFiles = returnPhotos + returnVideos
-            let expectedDownloads = self.compare(cloudFiles, second: facebookService.content)
+      // TODO: Compare the arrays and check for a change before starting the comparison algorithm
+      // TODO: Optimise this comparison for larger arrays
 
-            for item in expectedDownloads {
-              facebookService.download(item.url, platformType: .Facebook, fileName: item.name)
-            }
-          },
-          failure: {
-            // Videos were not able to load
-        })
-      },
-      failure: {
-        // Photos were not able to load
-      })
-  }
+      // Check for cloudFiles not in the localFiles array. Downloaded them if this is the case
+      for item in cloudFiles {
+        if !localFiles.contains({$0.name == item.name}) {
+          // Create the DB entry and download the file upon completion! I'm so excited about this!
+          DBController.sharedDBController.createFacebookFile(item, completion: { _ in
+            facebookService.download(item, platformType: PlatformType.Facebook, failure: { failedFile in
+              // If the download fialed then obliterate the file >:)
+              DBController.sharedDBController.deleteFacebookFileByName(failedFile.name)
+            })
+          })
+        }
+      }
 
-  // Compare two arrays returning files from the first array without a match in the second array
-  func compare(first: [CloudFile], second: [LocalFile]) -> [CloudFile] {
-    var result: [CloudFile] = []
+      // Check for localFiles not in the cloudFiles array. Delete them if this is the case
+      for item in localFiles {
+        if !cloudFiles.contains({$0.name == item.name}) {
+          guard let itemStringPath = facebookService.preferences.getRootPath(.Facebook) else {
+            continue
+          }
 
-    for one in first {
-      if !contains(second, item: one) {
-        result.append(one)
-      }
-    }
+          // Remove the DB entry
+          DBController.sharedDBController.deleteFacebookFileByName(item.name)
 
-    return result
-  }
+          // Remove the file from cachedLocalContent
+          if let removeIndex = facebookService.cachedLocalContent.indexOf({$0.name == item.name}) {
+            facebookService.cachedLocalContent.removeAtIndex(removeIndex)
 
-  func contains(array: [LocalFile], item: CloudFile) -> Bool {
-    for one in array {
-      if one.isEqual(item) {
-        return true
-      }
-    }
+            let itemPath = Path(itemStringPath)
+            let fileToDelete = File<NSDictionary>(path: itemPath)
 
-    return false
+            // Delete the file
+            do {
+              try fileToDelete.delete()
+            } catch FileKitError.DeleteFileFail(path: itemPath) {
+              ErrorController.sharedErrorController.getMainViewController().errorMessage.stringValue =
```

Comment:
Same thing here with the error messages not accessing the MainViewController directly. Make sure to change this everywhere.